### PR TITLE
[NeuralChat] Integrate PhotoAI backend into NeuralChat

### PR DESCRIPTION
**Pull request by letonghan**
## Type of Change

feature
API changed

## Description

Integrate PhotoAI backend services into NeuralChat.
JIRA: https://jira.devtools.intel.com/browse/NLPTOOLKIU-1007

## Expected Behavior & Potential Risk

The PhotoAI backend service will be able to start using neuralchat_server.

## How has this PR been tested?

local tested.

## Dependency Change?

Add dependencies for  retrieval.

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/examples/deployment/photo_ai/backend/run.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ export KMP_SETTINGS=1
 export KMP_AFFINITY=granularity=fine,compact,1,0
 
 # OMP
-export OMP_NUM_THREADS=56
```
Comment (kevinintel) [inline intel_extension_for_transformers/neural_chat/examples/deployment/photo_ai/backend/run.sh:27]: why 52?

Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/examples/deployment/photo_ai/backend/run.sh:27]: will change it back to 56.

**Review by lvliang-intel — APPROVED** [review]

File: .github/workflows/unit-test-neuralchat.yml [hunk 2]
```diff
@@ -103,7 +103,9 @@ jobs:
           && pip install yacs uvicorn optimum optimum[habana] \
           && pip install sentence_transformers unstructured markdown rouge_score \
           && pip install einops \
-          && pip install --upgrade accelerate"
+          && pip install --upgrade accelerate \
+          && pip install urllib3 \
```
Comment (VincyZhang) [inline .github/workflows/unit-test-neuralchat.yml:107]: need to add into neuralchat/tests/requirements.txt

Comment (letonghan) [reply .github/workflows/unit-test-neuralchat.yml:107]: added

**Comment by VincyZhang** [comment]
is coverage drop acceptable? 
image

**Pull request merged**